tests/generation/test_configuration_utils.py

```diff
@@ -679,18 +679,10 @@ def setUpClass(cls):
         cls._token = TOKEN
         HfFolder.save_token(TOKEN)
 
-    @staticmethod
-    def _try_delete_repo(repo_id, token):
-        try:
-            # Reset repo
-            delete_repo(repo_id=repo_id, token=token)
-        except:  # noqa E722
-            pass
-
     def test_push_to_hub(self):
+        repo_id = f"{USER}/test-generation-config"
         with tempfile.TemporaryDirectory() as tmp_dir:
-            try:
-                tmp_repo = f"{USER}/test-generation-config-{Path(tmp_dir).name}"
+            with TemporaryHubRepo(prefix=repo_id, random_id=Path(tmp_dir).name, token=self._token) as tmp_repo:
```

Comment:
just allow passing an (random) `id` to `TemporaryHubRepo`. I give the argument the name `random_id` as I want to emphasis it should be random, and it's the caller's responsibility to make sure it's random (if it is passed)